pex/resolver.py

```diff
@@ -247,16 +263,27 @@ def filter_packages_by_ttl(cls, packages, ttl, now=None):
     return [package for package in packages
         if package.remote or package.local and (now - os.path.getmtime(package.local_path)) < ttl]
 
+  @classmethod
+  def get_cache_key(cls, dist_location):
+    return os.stat(dist_location).st_size
```

Comment:
Rather than size, can we use something more robust like a content digest?

Looking at django 1.6 (which is ~7MB) which you profiled:

```
tmp$ time md5 Django-1.6.11-py2.py3-none-any.whl
MD5 (Django-1.6.11-py2.py3-none-any.whl) = e43d8c4147a821205b554aebc1da62d5

real	0m0.025s
user	0m0.019s
sys	0m0.008s
tmp$ time shasum -a 1 Django-1.6.11-py2.py3-none-any.whl
0d7d9071158294930ae86b776eb9ff9011eb1d7b  Django-1.6.11-py2.py3-none-any.whl

real	0m0.076s
user	0m0.047s
sys	0m0.021s
tmp$ time shasum -a 256 Django-1.6.11-py2.py3-none-any.whl
6f2cc848b2b72adf53a6f3f21be049c477e82c408bce7cedb57efeb0984bde24  Django-1.6.11-py2.py3-none-any.whl

real	0m0.100s
user	0m0.062s
sys	0m0.023s
```

I'm definitely happy to add 50ms per 7M dependency to pex building. I could also live with an extra 200ms.

Reply:
Thanks for taking a look! I agree that a hash digest would certainly be more robust. I changed it to size purely to squeeze out some perf, but it may make sense to switch back. I'll wait for some additional feedback on the perf impact.